services/blockchain-indexer/shared/dataService/business/events.js

```diff
@@ -63,14 +63,17 @@ const getEventsByHeight = async (height) => {
 	const cachedEvents = await eventCache.get(height);
 	if (cachedEvents) return JSON.parse(cachedEvents);
 
-	// Get from DB
-	const eventsTable = await getEventsTable();
-	const dbEventStrs = await eventsTable.find({ height }, ['eventStr']);
-
-	if (dbEventStrs.length) {
-		const dbEvents = dbEventStrs.map(({ eventStr }) => eventStr ? JSON.parse(eventStr) : eventStr);
-		await eventCache.set(height, JSON.stringify(dbEvents));
-		return dbEvents;
+	// Get from DB only when isPersistEvents is enabled
+	if (config.isPersistEvents) {
```

Comment:
Could you please ensure that with the following change we still persist the events for the non-finalized blocks?

Reply:
Discussed offline!